In [ ]:
!gdown 1urE8XVXevyOiowQ6uKssc75mlMWaOvaq

Downloading...
From: https://drive.google.com/uc?id=1urE8XVXevyOiowQ6uKssc75mlMWaOvaq
To: /content/delhivery_data.txt
100% 55.6M/55.6M [00:00<00:00, 64.2MB/s]


In [ ]:
import numpy as np
import pandas as pd
from scipy.stats import norm


In [ ]:
df=pd.read_csv('delhivery_data.txt')
df.head(1)

data          trip_creation_time  \
0  training  2018-09-20 02:35:36.476840   

                                 route_schedule_uuid route_type  \
0  thanos::sroute:eb7bfc78-b351-4c0e-a951-fa3d5c3...    Carting   

                 trip_uuid source_center                 source_name  \
0  trip-153741093647649320  IND388121AAA  Anand_VUNagar_DC (Gujarat)   

  destination_center               destination_name  \
0       IND388620AAB  Khambhat_MotvdDPP_D (Gujarat)   

                od_start_time  ...     cutoff_timestamp  \
0  2018-09-20 03:21:32.418600  ...  2018-09-20 04:27:55   

   actual_distance_to_destination  actual_time  osrm_time osrm_distance  \
0                        10.43566         14.0       11.0       11.9653   

     factor  segment_actual_time  segment_osrm_time  segment_osrm_distance  \
0  1.272727                 14.0               11.0                11.9653   

   segment_factor  
0        1.272727  

[1 rows x 24 columns]

In [ ]:
df.info()
#the trip creation,od_start_time,od_end_time,cutoff time stamp, all these columns should be converted to date time type

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 144867 entries, 0 to 144866
Data columns (total 24 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   data                            144867 non-null  object 
 1   trip_creation_time              144867 non-null  object 
 2   route_schedule_uuid             144867 non-null  object 
 3   route_type                      144867 non-null  object 
 4   trip_uuid                       144867 non-null  object 
 5   source_center                   144867 non-null  object 
 6   source_name                     144574 non-null  object 
 7   destination_center              144867 non-null  object 
 8   destination_name                144606 non-null  object 
 9   od_start_time                   144867 non-null  object 
 10  od_end_time                     144867 non-null  object 
 11  start_scan_to_end_scan          144867 non-null  float64
 12  is_cutoff       

In [ ]:
df.isnull().sum()

data                                0
trip_creation_time                  0
route_schedule_uuid                 0
route_type                          0
trip_uuid                           0
source_center                       0
source_name                       293
destination_center                  0
destination_name                  261
od_start_time                       0
od_end_time                         0
start_scan_to_end_scan              0
is_cutoff                           0
cutoff_factor                       0
cutoff_timestamp                    0
actual_distance_to_destination      0
actual_time                         0
osrm_time                           0
osrm_distance                       0
factor                              0
segment_actual_time                 0
segment_osrm_time                   0
segment_osrm_distance               0
segment_factor                      0
dtype: int64

### **Observation** source_name column has 293  null values and destination_name  column has 261 null values
